app/code/Magento/Sitemap/Model/Observer.php

```diff
@@ -112,9 +123,17 @@ public function scheduledGenerateSitemaps()
         foreach ($collection as $sitemap) {
             /* @var $sitemap \Magento\Sitemap\Model\Sitemap */
             try {
+                $this->appEmulation->startEnvironmentEmulation(
+                    $sitemap->getStoreId(),
+                    'frontend',
```

Comment:
Please use the constant

Reply:
This Class have many dependency, so if i will use constant the codeMess test will fail.

Reply:
In this case it's better to decompose the class in favour to dropping constant usage